src/property.h

```diff
@@ -40,91 +40,41 @@
 
 namespace nttec {
 
-/** The location of a value in a buffer. */
-struct ValueLocation {
-    off_t offset;
-    uint32_t fragment_id;
-
-    /** Construct an invalid location. */
-    ValueLocation() : offset(-1), fragment_id(0) {}
-
-    /** Construct a location from an offset and a fragment ID. */
-    ValueLocation(off_t pos, uint32_t fragment)
-        : offset(pos), fragment_id(fragment)
-    {
-    }
-
-    /** Construct a location from its string representation. */
-    explicit ValueLocation(const std::string& str);
-
-    /**  Checks if the two locations are equal. */
-    bool operator==(const ValueLocation& other) const
-    {
-        return offset == other.offset && fragment_id == other.fragment_id;
-    }
-
-    /** Return the string representation of the location. */
-    std::string to_string() const
-    {
-        std::string str(std::to_string(offset));
-        str += ':';
-        str += std::to_string(fragment_id);
-        return str;
-    }
-
-    inline off_t get_offset() const
-    {
-        return offset;
-    }
-};
-
-} // namespace nttec
-
-namespace std {
-
-template <>
-struct hash<nttec::ValueLocation> {
-    std::size_t operator()(const nttec::ValueLocation& k) const
-    {
-        return std::hash<off_t>()(k.offset)
-               + std::hash<uint32_t>()(k.fragment_id);
-    }
-};
-
-} // namespace std
-
-namespace nttec {
-
 /** Ancillary data attached to values.
  *
  * A property carries extra-information (whose interpretation is left to the
  * reader) related to a specific value (identified by its location).
+ * It wraps a map whose each element is a key/value where
+ *  - key indicates the location of symbol whose value should be adjusted
+ *  - value indicates value that could be used to adjust the symbol value
+ * For prime fields, value is always 1.
+ * For NF4, value is an uint32_t integer.
  */
 class Properties {
   public:
-    inline void add(const ValueLocation& loc, const std::string& data)
+    inline void add(const off_t loc, const uint32_t data)
     {
         props[loc] = data;
     }
 
-    inline const std::string* const get(const ValueLocation& loc) const
+    inline const uint32_t get(const off_t loc) const
```

Comment:
I think you can remove the const here, since you return by value.

Reply:
it's done